internal/oci/verifier.go

```diff
@@ -40,9 +42,13 @@ type Verifier interface {
 
 // options is a struct that holds options for verifier.
 type options struct {
-	PublicKey  []byte
-	ROpt       []remote.Option
-	Identities []cosign.Identity
+	PublicKey   []byte
+	ROpt        []remote.Option
+	Identities  []cosign.Identity
+	TrustPolicy *trustpolicy.Document
+	Auth        authn.Authenticator
+	Keychain    authn.Keychain
+	Insecure    bool
```

Comment:
I was the one that suggested a unified `options` in order too many functional options.
> It may be better to just keep the Verifier interface in this package and have two separate subpackages for cosign and notation which implement the Verifier interface.

This is  a good idea and would make the code more lisible.

Reply:
I've moved `cosign` to its own package in [b3ed2d1](https://github.com/fluxcd/source-controller/pull/1075/commits/b3ed2d1befd95aa14d6f4608b55dbfc0eb53fd30) and `notation` in [851e012](https://github.com/fluxcd/source-controller/pull/1075/commits/851e012822227b46c07bdda9f616258fcd1e1f42) and [06b50ac](https://github.com/fluxcd/source-controller/pull/1075/commits/06b50ac5269d3ad7f8e668b2e9acb90b6e8b39ed). I've also updated some of the func names and structs in [c3e3e19](https://github.com/fluxcd/source-controller/pull/1075/commits/c3e3e193ea6f3114587b9f623725138dcd54d3f1).